### PR TITLE
Upgrades: Nudge global test

### PR DESCRIPTION
**Pull request by artpi**
- Introduces global abtest for all nudges
- refactors some areas of #5151 

CC @mtias @rralian

### CODE REVIEW

**Comment by gwwar** [comment]
@artpi We can avoid the componentDidMount code if we use a `TrackComponentView` component in the render method. See: https://github.com/Automattic/wp-calypso/blob/master/client/lib/analytics/track-component-view/README.md

**Comment by artpi** [comment]
Tracking fix moved to #5151 
Closing this PR

File: client/lib/abtest/active-tests.js [hunk 1]
```diff
@@ -63,6 +63,15 @@ module.exports = {
 		},
 		defaultVariation: 'namegen'
 	},
+	nudges: {
+		datestamp: '20160101',
+		variations: {
+			showAll: 100,
```
Comment (artpi) [inline client/lib/abtest/active-tests.js]: My intention is to merge this, finish existing tests and then ramp this up

**Comment by mtias** [comment]
This needs a rebase.

**Pull request merged**